Setup

bitdivine opened the pull request:
# Motivation

<!-- Describe the motivation that lead to the PR -->

# Changes

<!-- List the changes that have been developed -->

# Tests

<!-- Please provide any information or screenshots about the tests that have been done -->

bitdivine commented:
thank you for the testing @lmuntanerDfinity

lmuntanerDfinity reviewed (changes requested):
Added the comments that we discovered when running this

bitdivine commented:
I have:
- Switched from .bashrc to .profile as the script and added a dialogue to check with the user if that is OK
- Added a step to install wget
- Filled in the step to install cmake.

bitdivine commented on scripts/setup:
```diff
@@ -0,0 +1,152 @@
+
+OS="$(uname | tr '[:upper:]' '[:lower:]')"
+
+install_rosetta_maybe
```
Already done. :-) Wget is installed first.  This is a function, it doesn't get called until invoked at the end of the script.

bitdivine replied:
Actually reverted, as rosetta and brew need to be installed before wget.

bitdivine commented on scripts/setup:
```diff
@@ -0,0 +1,152 @@
+have_correct_dfx || install_dfx
+command -v gtar || tar --help | grep GNU >/dev/null || "install_tar_$OS"
+command -v flutter || "install_flutter"
+command -v cmake || "install_cmake_$OS"
```
Already done. :-)

bitdivine replied:
Ah, push failed.  Pushed again.

lmuntanerDfinity commented:
Let me run it again afterwards

bitdivine commented:
Run on a new user:
- Needed sudo and brew.  Added checks for these and an installer for brew.

lmuntanerDfinity reviewed (approved):
Thanks!!

bitdivine commented:
Created another new user and reran.  Repeated until the whole process was smooth on M1.

Next steps:
- Final modifications to ./deploy.sh in the respective PR
- Update README.md in a new PR
